corporate/tests/test_stripe.py

```diff
@@ -128,6 +128,8 @@ def test_initial_upgrade(self, mock_create_subscription: mock.Mock,
         self.assert_in_success_response(['We can also bill by invoice'], response)
         self.assertFalse(user.realm.has_seat_based_plan)
         self.assertNotEqual(user.realm.plan_type, Realm.PREMIUM)
+        user.realm.message_visibility_limit = 10000
```

Comment:
We should remove the changes to this file, and have another test in `zerver/tests` just for do_change_plan_type, since that's going to eventually be a potentially complicated function with lots of things changing.

I guess one thing we could do is mock do_change_plan_type to make sure it is called. (To prevent future mistakes of someone just setting the Realm.plan_type). But not worth it if that's hard.